curvefs/proto/metaserver.proto

```diff
@@ -114,6 +114,10 @@ message CreateDentryRequest {
     required uint32 copysetId = 2;
     required uint32 partitionId = 3;
     required Dentry dentry = 4;
+    required bool enableSumInDir = 5;
```

Comment:
1. required need change to optional for compatible upgrade.
2. enableSumInDir is not needed and update to parent inode directly, the function enablesumInDir will be discarded.
3. new file length is 0 and length here is not needed

Reply:
For FuseOpLink And FuseOpSymLink, length is not zero, So length is needed?